Jamulus.pro

```diff
@@ -59,6 +64,10 @@ TRANSLATIONS = src/translation/translation_de_DE.ts \
     src/translation/translation_sv_SE.ts \
     src/translation/translation_zh_CN.ts
 
+# This value must match the value for DirIter in the function
```

Comment:
Please write briefly what the value means and not just say "this must equal that"

Reply:
Also, an explanation of why that function _cannot_ use the variable here would be good.

Reply:
Not sure how much detailed explanation belongs in the `Jamulus.pro` file, but I'll explain the issue here and you guys can advise.

In `src/util.cpp` is the following function excerpt:

```
QMap<QString, QString> CLocale::GetAvailableTranslations()
{
    QMap<QString, QString> TranslMap;
    QDirIterator           DirIter ( ":/translations" );
    ......
```

The string `:/translations` is a reference to internal resources, currently defined in `src/resources.qrc`:

```
<RCC>
    <qresource prefix="/translations">
        <file alias="translation_de">translation/translation_de_DE.qm</file>
    </qresource>
    <qresource prefix="/translations">
        <file alias="translation_fr">translation/translation_fr_FR.qm</file>
    </qresource>
    ......
```

It is the value of `prefix` in this file that is referred to by `DirIter` and must therefore match.

Now, when we move to `embed_translations`, the translations are put in a new resource file generated by `qmake`, viz. `qmake_qmake_qm_files.qrc`. By default, `qmake` gives them the prefix `i18n`:

```
<!DOCTYPE RCC><RCC version="1.0">
<qresource prefix="i18n">
<file alias="translation_de_DE.qm">/home/tony/jamulus/.qm/translation_de_DE.qm</file>
<file alias="translation_fr_FR.qm">/home/tony/jamulus/.qm/translation_fr_FR.qm</file>
......
```

So, to be able to find them, exactly one of two things needs to be done:
1. Modify the value of `DirIter` in `CLocale::GetAvailableTranslations()` to contain `:/i18n` instead of `:/translations`.
2. Tell `qmake` to use a prefix of `translations` instead of the default, by setting `QM_FILES_RESOURCE_PREFIX` in `Jamulus.pro`, to match the existing value for `DirIter`.

I did the latter, and added a comment to alert the user of the consistency needed between the two places. But I didn't feel it needed an essay at that point.

Maybe instead we revert to option 1 above instead? Maybe with a short comment about the default value used by `qmake` for the resource. Or else stick with option 2 and in `Jamulus.pro` add a link to this comment for further details? Opinions? After typing this, I'm now leaning towards option 1. Especially since that function already has another change within it.

Reply:
If there's a variable that can affect what qmake does that affects what the code needs to do, then the _code_ should be sensitive to that value, too.  So I'd still say we should use `QM_FILES_RESOURCE_PREFIX` in the code, defaulting to the qmake default if it's not set.

Reply:
This would require passing in the value of `QM_FILES_RESOURCE_PREFIX` to the compiler as a define. I disagree with the need in this case. For something that just needs to be got right once and there is never going to be a need to change, I think that is overkill. I would agree if it were something the user needs the ability to choose or change, but that's not the case here.

So my preferred option is to take out the `QM_FILES_RESOURCE_PREFIX` change from `Jamulus.pro` and just accept the default used by `qmake`, and to update the function in `util.cpp` to use the default prefix value, with an explanatory comment. I'll add a commit to that effect for review, which I can squash into the original commit before merging.